test/PreAMMBatcher.spec.ts

```diff
@@ -0,0 +1,105 @@
+import UniswapV2Factory from '../node_modules/@uniswap/v2-core/build/UniswapV2Factory.json';
+
+import ERC20 from '../node_modules/@openzeppelin/contracts/build/contracts/ERC20Mintable.json';
+import {Order, DOMAIN_SEPARATOR} from '../src/js/orders.spec';
+import BN from 'bn.js';
+
+use(solidity);
+
+describe('PreAMMBatcher', () => {
+  const [walletDeployer, walletTrader1, walletTrader2] = new MockProvider().getWallets();
+  let batcher: Contract;
+  let token0: Contract;
+  let token1: Contract;
+  let uniswapPair: Contract;
+  let uniswapFactory: Contract;
+  let uniswapPairAddress: string;
+  const initial_uniswap_funding_token0 = utils.parseEther('10');
+  const initial_uniswap_funding_token1 = utils.parseEther('10');
+
+  const mockRelevantTokenDataForPreBatchTrade = async (sellOrder1: Order, sellOrder2: Order) => {
+    // todo: mock actual right information
+    // mock un-relevant data to make preBatchTrade call pass
+    await token0.mock.transferFrom.returns(true);
+    await token1.mock.transferFrom.returns(true);
+    await token0.mock.transfer.returns(true);
+    await token1.mock.transfer.returns(true);
+    await token0.mock.balanceOf.withArgs(sellOrder1.wallet.address).returns(sellOrder1.sellAmount);
+    await token1.mock.balanceOf.withArgs(sellOrder2.wallet.address).returns(sellOrder2.sellAmount);
+    await token0.mock.balanceOf.withArgs(batcher.address).returns('1');
+    await token1.mock.balanceOf.withArgs(batcher.address).returns('1');
+    await token0.mock.balanceOf.withArgs(uniswapPair.address)
+      .returns(new BN(initial_uniswap_funding_token0.toString()).add(sellOrder1.sellAmount).toString());
+    await token1.mock.balanceOf.withArgs(uniswapPair.address).returns(initial_uniswap_funding_token1);
+  };
+
+  beforeEach(async () => {
+    token0 = await deployMockContract(walletDeployer, ERC20.abi);
+    token1 = await deployMockContract(walletDeployer, ERC20.abi);
+    uniswapFactory = await deployContract(walletDeployer, UniswapV2Factory, [walletDeployer.address]);
+    await uniswapFactory.createPair(token0.address, token1.address, {gasLimit: 6000000});
+    uniswapPairAddress = await uniswapFactory.getPair(token0.address, token1.address);
+    uniswapPair = await deployContract(walletDeployer, UniswapV2Pair);
+    uniswapPair = await uniswapPair.attach(uniswapPairAddress);
+    batcher = await deployContract(walletDeployer, PreAMMBatcher, [uniswapFactory.address]);
+
+    await token0.mock.balanceOf.withArgs(uniswapPair.address).returns(initial_uniswap_funding_token0);
+    await token1.mock.balanceOf.withArgs(uniswapPair.address).returns(initial_uniswap_funding_token1);
+
+    await uniswapPair.mint(walletDeployer.address, {gasLimit: 500000});
+    expect((await uniswapPair.getReserves())[0]).to.equal(initial_uniswap_funding_token0);
+  });
+
+  it('DOMAIN_SEPARATOR is correct', async () => {
+    expect(await batcher.DOMAIN_SEPARATOR()).to.equal(DOMAIN_SEPARATOR);
```

Comment:
Not very fond of this weird new way of making assertions... what next `expect(x).is.less.than.or.equal.to(y)`? or 
`expect("my text").contains.the.letter("x")`

Reply:
I love chai matchers